src/openzaak/components/autorisaties/api/serializers.py

```diff
@@ -2,18 +2,83 @@
 # Copyright (C) 2020 Dimpact
 import logging
 
+from django.conf import settings
 from django.db import transaction
+from django.urls import reverse
 
-from vng_api_common.authorizations.models import Applicatie
+from vng_api_common.authorizations.models import Applicatie, Autorisatie
 from vng_api_common.authorizations.serializers import (
     ApplicatieSerializer as _ApplicatieSerializer,
+    AutorisatieBaseSerializer,
 )
+from vng_api_common.constants import ComponentTypes
 from vng_api_common.models import JWTSecret
 
+from openzaak.utils import build_absolute_url
+
 logger = logging.getLogger(__name__)
 
 
 class ApplicatieSerializer(_ApplicatieSerializer):
+    def to_representation(self, instance):
```

Comment:
I agree that in the second and third case, we should block updating/adding individual autorisaties if there are already CatalogusAutorisaties for the corresponding catalogus.

In the last case, where the zaaktypen are not part of a Catalogus for which the Applicatie has a CatalogusAutorisatie, I don't think we have to block it, I think that should still be allowed.

Maybe we should also change the admin form to apply this same check the other way around? E.g. if you have Autorisaties for Zaaktypen with Catalogus A and you try to add a CatalogusAutorisatie for Catalogus A? Or am I making things too complicated then?

Reply:
Ok, let's allow the last case, and here the CatalogusAutorisatie would be deleted, right?

Reply:
Ah I misread the first time, I added validation to the serializer to check if there is overlap between the CatalogusAutorisaties and the typen you try to create autorisaties for via the API. 

But as I read it now, you'd rather avoid this validation and just drop the CatalogusAutorisaties (so you either manage your (catalogus)autorisaties via the admin or you manage your autorisaties via the API). Do I understand that correctly?

Reply:
Yes! We can't manage CatalogusAutorisaties in the API, so when API request has typen both individual and from existing CatalogusAutorisaties it can cause some problems with inconsistency of authorizations. 

In PUT case when you have some individual `Autorisaties` which are not present in existing `CatalogusAutorisaties` we can drop `CatalogusAutorisaties`, right? (Or just raise a validation error to tell client to manage authorizations in the admin)

Reply:
I guess dropping them in that case is probably better. And in case of PATCH we could keep them?

Reply:
I think in PATCH we still change the whole `autorisaties` attribute, so it should be the same as PUT
We will just keep attributes not included in request body as they were

Reply:
Modified the PUT/PATCH behavior: https://github.com/open-zaak/open-zaak/pull/1706/files#diff-d85506e1b315c329356c2f360759d31c4d90723f4ce1e2169dbab1312a3dfb4eR107-R116